rotkehlchen/tests/unit/decoders/test_curve.py

```diff
@@ -2277,3 +2277,146 @@ def test_vote_escrow_withdraw(ethereum_transaction_decoder, ethereum_accounts):
         ),
     ]
     assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('load_global_caches', [[CPT_CURVE]])
+@pytest.mark.parametrize('gnosis_accounts', [['0xc37b40ABdB939635068d3c5f13E7faF686F03B65']])
+def test_gauge_deposit_and_stake(database, gnosis_inquirer, gnosis_accounts, load_global_caches):
+    tx_hash = deserialize_evm_tx_hash('0x52141e28c55744cdfddaa1ab5614642f61459d7624a51981e21ec06176c8f65c')  # noqa: E501
+    timestamp, gauge_address, deposited_amount, gas_fees, received_amount = TimestampMS(1717744880000), string_to_evm_address('0xd91770E868c7471a9585d1819143063A40c54D00'), '834.517942394847767492', '0.0023483922', '431.905365995549947348'  # noqa: E501
+    get_or_create_evm_token(  # gauge token should already exist in db by reloading cache tokens
+        userdb=database,
+        evm_address=gauge_address,
+        chain_id=gnosis_inquirer.chain_id,
+        token_kind=EvmTokenKind.ERC20,
+        evm_inquirer=gnosis_inquirer,
+        protocol=CURVE_POOL_PROTOCOL,
+    )
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=gnosis_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+        load_global_caches=load_global_caches,
+    )
+    assert events == [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.GNOSIS,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_XDAI,
+            balance=Balance(amount=FVal(gas_fees)),
+            location_label=gnosis_accounts[0],
+            notes=f'Burned {gas_fees} XDAI for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.GNOSIS,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=Asset('eip155:100/erc20:0xcB444e90D8198415266c6a2724b7900fb12FC56E'),
+            balance=Balance(amount=FVal(deposited_amount)),
+            location_label=gnosis_accounts[0],
+            notes=f'Deposit {deposited_amount} EURe into {gauge_address} curve gauge',
```

Comment:
Question here: Should we have a different action verb for Deposit vs Deposit and stake? So "Deposit and stake" here? Or is that implied by the fact it's a curve gauge we are depositng to, so it's actually staking?

Reply:
I believe the stake is implicit since we deposit to a gauge